common/search/src/test/scala/weco/api/search/elasticsearch/ElasticsearchServiceTest.scala

```diff
@@ -82,15 +88,176 @@ class ElasticsearchServiceTest
     }
   }
 
+  describe("findById") {
+    it("finds documents by ID") {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val thingToQueryFor = thingsToIndex.head
+
+        val findByIdFuture =
+          elasticsearchService.findById[ExampleThing](thingToQueryFor.id)(index)
+
+        whenReady(findByIdFuture) { findByIdEither =>
+          val foundById = findByIdEither.right.value
+          foundById shouldBe thingToQueryFor
+        }
+      }
+    }
+
+    it("should return an appropriate error if no ID matches") {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val badId = createCanonicalId
+
+        val findByIdFuture =
+          elasticsearchService.findById[ExampleThing](badId)(index)
+
+        whenReady(findByIdFuture) { findByIdEither =>
+          val failedFindByID = findByIdEither.left.value
+
+          failedFindByID shouldBe a[DocumentNotFoundError[_]]
+        }
+      }
```

Comment:
Personally I find it nicer when I can bin the intermediate variables, but it's not super important:

```suggestion
        whenReady(findByIdFuture) {
          _.left.value shouldBe a[DocumentNotFoundError[_]]
        }
      }
```